test/cmd/authentication.sh

```diff
@@ -17,39 +17,39 @@ os::cmd::expect_success 'oadm policy add-role-to-user admin scoped-user'
 # initialize the user object
 os::cmd::expect_success 'oc login -u scoped-user -p asdf'
 os::cmd::expect_success 'oc login -u system:admin'
-username=$(oc get user/scoped-user -o jsonpath={.metadata.name})
-useruid=$(oc get user/scoped-user -o jsonpath={.metadata.uid})
+username="$(oc get user/scoped-user -o jsonpath='{.metadata.name}')"
+useruid="$(oc get user/scoped-user -o jsonpath='{.metadata.uid}')"
 os::cmd::expect_success_and_text "oc policy can-i --list -n cmd-authentication --as=scoped-user" 'get.*pods'
 
-whoamitoken=$(oc process -f ${OS_ROOT}/test/testdata/authentication/scoped-token-template.yaml TOKEN_PREFIX=whoami SCOPE=user:info USER_NAME="${username}" USER_UID="${useruid}" | oc create -f - -o name | awk -F/ '{print $2}')
-os::cmd::expect_success_and_text 'oc get user/~ --token="${whoamitoken}"' "${username}"
-os::cmd::expect_failure_and_text 'oc get pods --token="${whoamitoken}" -n cmd-authentication' 'prevent this action; User "scoped-user" cannot list pods in project "cmd-authentication"'
+whoamitoken="$(oc process -f "${OS_ROOT}/test/testdata/authentication/scoped-token-template.yaml" TOKEN_PREFIX=whoami SCOPE=user:info USER_NAME="${username}" USER_UID="${useruid}" | oc create -f - -o name | awk -F/ '{print $2}')"
```

Comment:
Yes. Everything inside of `$()` is interpreted in a 'clean' scope -- as if you opened the subshell and typed it there. `$()` is also nestable. Don't get any ideas.